src/physics/site_types/spinhalf.jl

```diff
@@ -22,47 +22,56 @@ function state(::SpinHalfSite,
   return 0
 end
 
-function op(::SpinHalfSite,
-            s::Index,
-            opname::AbstractString;kwargs...)::ITensor
-  Up = s(1)
-  UpP = s'(1)
-  Dn = s(2)
-  DnP = s'(2)
- 
-  Op = emptyITensor(s',dag(s))
 
-  if opname == "S⁺" || opname == "Splus" || opname == "S+"
-    Op[UpP, Dn] = 1.
-  elseif opname == "S⁻" || opname == "Sminus" || opname == "S-"
-    Op[DnP, Up] = 1.
-  elseif opname == "Sˣ" || opname == "Sx"
-    Op[UpP, Dn] = 0.5
-    Op[DnP, Up] = 0.5
-  elseif opname == "iSʸ" || opname == "iSy"
-     Op[UpP, Dn] = 0.5
-     Op[DnP, Up] = -0.5
-  elseif opname == "Sʸ" || opname == "Sy"
-     Op = complex(Op) 
-     Op[UpP, Dn] = -0.5*im
-     Op[DnP, Up] = 0.5*im
-  elseif opname == "Sᶻ" || opname == "Sz"
-     Op[UpP, Up] = 0.5
-     Op[DnP, Dn] = -0.5
-  elseif opname == "projUp"
-     Op[UpP, Up] = 1.
-  elseif opname == "projDn"
-    Op[DnP, Dn] = 1.
-  elseif opname == "Up" || opname == "↑"
-    pU = emptyITensor(s)
-    pU[Up] = 1.
-    return pU
-  elseif opname == "Dn" || opname == "↓"
-    pD = emptyITensor(s)
-    pD[Dn] = 1.
-    return pD
-  else
-    throw(ArgumentError("Operator name '$opname' not recognized for SpinHalfSite"))
-  end
-  return Op
+function op!(::TagType"S=1/2",
+             ::Union{OpName"Sz"},
+             Op::ITensor,
+             s::Index)
+  Op[s'=>1, s=>1] = 0.5
+  Op[s'=>2, s=>2] = -0.5
+end
+
+function op!(::TagType"S=1/2",
+             ::Union{OpName"S+",OpName"Splus"},
+             Op::ITensor,
+             s::Index)
+  Op[s'=>1, s=>2] = 1.0
+end
+
+function op!(::TagType"S=1/2",
+             ::Union{OpName"S-",OpName"Sminus"},
```

Comment:
I think I would prefer using the strategy you use down below of having two functions (one which calls the other), instead of using `Union`, but I don't care too much either way. It seems a bit more "extensible" that way, like it makes it a bit clearer how someone would give their own name for an operator.